boefjes/boefjes/plugins/kat_cve_2023_35078/normalize.py

```diff
@@ -0,0 +1,75 @@
+from typing import Iterable, List, Union
+
+from boefjes.job_models import NormalizerMeta
+from octopoes.models import OOI, Reference
+from octopoes.models.ooi.findings import CVEFindingType, Finding
+from octopoes.models.ooi.software import Software, SoftwareInstance
+from packaging import version
+
+PATCHED_VERSIONS: List[str] = ["11.10.0.2", "11.9.1.1", "11.8.1.1"]
+
+
+def extract_js_version(html_content: str) -> Union[version.Version, bool]:
+    telltale = "/mifs/scripts/auth.js?"
+    telltale_position = html_content.find(telltale)
+    if telltale_position == -1:
+        return False
+    version_end = html_content.find('"', telltale_position)
+    version_string = html_content[telltale_position + len(telltale) : version_end]
+    if not version_string:
+        return False
+    return version.parse(" ".join(strip_vsp_and_build(version_string)))
+
+
+def extract_css_version(html_content: str) -> Union[version.Version, bool]:
+    telltale = "/mifs/css/windowsAllAuth.css?"
+    telltale_position = html_content.find(telltale)
+    if telltale_position == -1:
+        return False
+    version_end = html_content.find('"', telltale_position)
+    version_string = html_content[telltale_position + len(telltale) : version_end]
+    if not version_string:
+        return False
+    return version.parse(" ".join(strip_vsp_and_build(version_string)))
+
+
+def strip_vsp_and_build(url: str) -> Iterable[str]:
+    url_parts = url.split()
+    for part in url_parts:
+        if str(part).lower() == "vsp":
+            continue
+        if str(part).lower() == "build":
+            break
+        yield part
+
+
+def check_for_versions(patched_versions: List[version.Version], detected_versions: List[version.Version]) -> bool:
```

Comment:
Maybe name this `is_vulnerable_version`? I had to check the code whether it returns True or False if it is vulnerable, it might be good idea to make this clear in the name of the function.